From 800354c46f18478a1c7d9f96eda105a25c7a6923 Mon Sep 17 00:00:00 2001
From: Ky-Anh Huynh <kyanh@theslinux.org>
Date: Sat, 6 Apr 2013 06:24:19 +0000
Subject: [PATCH] Replace the use of :run-parts by bash script

---
 .../sbin/update-ca-certificates                      | 20 ++++++++++++++------
 1 file changed, 14 insertions(+), 6 deletions(-)

diff --git a/ca-certificates/src/ca-certificates-20130119/sbin/update-ca-certificates b/ca-certificates/src/ca-certificates-20130119/sbin/update-ca-certificates
index 5375950..4a43e20 100755
--- a/ca-certificates/src/ca-certificates-20130119/sbin/update-ca-certificates
+++ b/ca-certificates/src/ca-certificates-20130119/sbin/update-ca-certificates
@@ -4,7 +4,7 @@
 #
 # Copyright (c) 2003 Fumitoshi UKAI <ukai@debian.or.jp>
 # Copyright (c) 2009 Philipp Kern <pkern@debian.org>
-# 
+#
 # This program is free software; you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
 # the Free Software Foundation; either version 2 of the License, or
@@ -152,12 +152,20 @@ HOOKSDIR=/etc/ca-certificates/update.d
 echo -n "Running hooks in $HOOKSDIR...."
 VERBOSE_ARG=
 [ "$verbose" = 0 ] || VERBOSE_ARG=--verbose
-eval run-parts $VERBOSE_ARG --test -- $HOOKSDIR | while read hook
-do
-  ( cat $ADDED
-    cat $REMOVED ) | $hook || echo E: $hook exited with code $?.
+
+# KyAnh 2013-April-6th:
+# Because we don't use run-parts, we need to replace the old code by
+# the new one. Please note the old code uses `--test` to list all
+# executable hooks, and within that option the `verbose` directive
+# (if any) has no effect.
+#
+find "$HOOKSDIR/" -mindepth 1 -maxdepth 1 -type f \
+| while read hook; do
+  if [[ -x "$hook" ]]; then
+    ( cat $ADDED
+      cat $REMOVED ) | $hook || echo E: $hook exited with code $?.
+  fi
 done
 echo "done."
 
 # vim:set et sw=2:
-
-- 
1.8.2

